Fixes being able to resist while dead.

phil235 opened the pull request:
- Fixes #15862
- Change some "for" loop with grabbed_by to be typeless since grabbed_by only contains grabs.
- Replace super weird "locate(/obj/item/weapon/grab, locate(/obj/item/weapon/grab, mob.grabbed_by.len))" by simple "if(grabbed_by.len)". This fixes Process_Grab(). When grabbed by someone, you can now break the grab by moving (success rate depends on grab state).
- Cleans up Process_Grab to be easier to read.
- Simplifies MobBump() proc, especially the part about pushing a mob that is pulling or is pulled/grabbed. We no longer check for mobs in a 1 tile radius every time we bump a mob.
- Fixes runtime with clear_reagents_to_vomit_pool() due to a typo.
- I refactored how a mob resisting out of a closet that is itself inside something is done.
- Fixes not being able to resist out of an unlocked unwelded locker. Now both moving and the resist button  work (except for the cardboard box). Fixes #14593
- You can no longer spam breakingout message out of a closet by moving while inside it. Fixes #15721
- Wrapping and unwrapping a locker no longer unwelds it magically. Fixes #14839
- I refactored closet/crate/item package wrapping.
- You can now wrap an item that is in your hand, and wrapping an item puts the created package in you hand. Also, opening a package puts the item in your hand instead of being dropped on the ground.
- <s>You can packagewrap belts and other storage items in which the package wrap item doesn't fit.(it does currently have the unintended side effect of giving you the "doesn't fit in X" message when wrapping there storage items though).</s>
- I moved some code around to where they should be.

phil235 committed:
Change some for loop with grabbed_by to be typeless since grabbed_by only contains grabs.
Replace super weird "locate(/obj/item/weapon/grab, locate(/obj/item/weapon/grab, mob.grabbed_by.len))" by simple if(grabbed_by.len)"
Cleans up Process_Grab to be easier to read.
Simplifies MobBump() proc, especially the part about pushing a mob that is pulling or is pullsed/grabbed. We no longer check for mobs in a 1 tile radius every time we bump a mob.
Fixes runtime with clear_reagents_to_vomit_pool() due to a typo.

phil235 committed:
I refactored how a mob resisting out of a closet that is itself inside something is done.
Fixes not being able to resist out of an unlocked unwelded locker. Now both moving and the resist button  work (except for the cardboard box).
You can no longer spam breakingout message ouf of a closet by moving while inside.
Wrapping and unwrapping a locker no longer unwelds it magically.
I refactored closet/crate/item package wrapping.
You can packagewrap belts and other storage items in which the package wrap item doesn't fit.(it does currently have the unintended side effect of giving you the "doesn't fit in X" message when wrapping there storage items though).

phil235 commented on code/game/objects/items/weapons/storage/secure.dm:
```diff
@@ -60,7 +60,7 @@
 
 	// -> storage/attackby() what with handle insertion, etc
-	..()
+	return ..()
```
all these storage items' attackby are supposed to return 1 when the item insertion succeeds.

RemieRichards commented:
@phil235 you should probably mention that ```Process_Grab()``` actually working is a pretty big mechanical change, people can now (attempt to) break grabs by moving.

phil235 commented:
@RemieRichards Process_Grab was broken?

RemieRichards commented:
@phil235 Hella broken, as far as I'm aware you can't locate a grab object inside another grab object INSIDE OF A NUMBER.

and since the entire proc was under that if, none of it ran.

phil235 commented:
Oh, true. I'll add it to the OP.

phil235 commented on code/modules/recycling/sortingmachinery.dm:
```diff
@@ -48,37 +42,48 @@
+	if(istype(loc, /atom/movable))
+		var/atom/movable/AM = loc //can't unwrap the wrapped container if it's inside something.
+		AM.relay_container_resist(user, O)
+		return
```
Currently this is only used for wrapped closet inside ripley mech's cargo.

phil235 committed:
Fixes some mistakes.
Updating the package wrap's descriptions to be accurate and useful.

phil235 commented on code/game/objects/structures/crates_lockers/closets.dm:
```diff
@@ -310,16 +312,25 @@
-	user.visible_message("<span class='warning'>[src] begins to shake violently!</span>",
-						"<span class='notice'>You lean on the back of [src] and start pushing the door open.</span>")
+	user << "<span class='notice'>You lean on the back of [src] and start pushing the door open.</span>"
+	visible_message("<span class='warning'>[src] begins to shake violently!</span>")
```
We have to give the visible_message to the closet, not the user, because otherwise noone but the user would get it.
I had to give the user his own message to inform them of their action, so they end up seeing two messages but I think it's worth it.

phil235 commented on code/modules/recycling/sortingmachinery.dm:
```diff
@@ -48,37 +42,48 @@
+		AM.relay_container_resist(user, O)
+		return
+	user << "<span class='notice'>You lean on the back of [O] and start pushing to rip the wrapping around it.</span>"
+	if(do_after(user, 50, target = O))
```
only takes 5 seconds to rip the paper. I didn't want to punish people in a locked and wrapped closet too much.

phil235 committed:
derp

TrustyGun commented:
average phil pr: 
"Fixes X" then in the desc "Refactored Y, Changed how Z works, etc etc"

PKPenguin321 commented:
>When grabbed by someone, you can now break the grab by moving (success rate depends on grab state).

Can you break out of neckgrabs because if so this isn't muy bien

phil235 commented:
@PKPenguin321 Yes, 5% chance per move. https://github.com/tgstation/-tg-station/pull/15899/files#diff-99ebcf7ba775c1dfd1e8fd98cdbebe11R238

PKPenguin321 commented:
Can we make it not do that?

phil235 commented:
@PKPenguin321 I'm not touching any grab functionality in this PR, I'm just fixing what was broken.

KazeEspada commented:
:+1: Fixing Grabs/Nerfing Grabs. Neck Grabs were shit.

KazeEspada commented on code/game/objects/items/stacks/wrap.dm:
```diff
@@ -0,0 +1,100 @@
+		new /obj/item/weapon/c_tube(get_turf(src))
+	return ..()
+
+/obj/item/weapon/c_tube
```
change to the trash type?

phil235 replied:
/obj/item/trash does nothing codewise, so I don't really see the point.

phil235 committed:
Some adjustments to wrapping code. I removed my possibility to wrap non-box storage item (that was at the expense of seeing a warning message that the packagewrap doesn't fit in the storage item every time you wrap one). So we're back to only wrapping boxes.
You can now wrap an item that is in your hand, and wrapping an item puts the created package in you hand.
Also, opening a package puts the item in your hand instead of being dropped on the ground.

phil235 commented:
Alright, I removed the questionable part about wrapping storage items other than boxes, which would give you a weird warning message. We're back to just wrapping non storage items and boxes.

Also:
You can now wrap an item that is in your hand, and wrapping an item puts the created package in you hand. Opening a package puts the item in your hand instead of being dropped on the ground.
